src/app/filecoin-plus/allocators/utils/createUniqueOptionsFromData.ts

```diff
@@ -0,0 +1,34 @@
+import type { Allocator } from '../schemas/AllocatorSchema'
+
+type AllocatorStringValues = Omit<Allocator, 'application'>
+
+export function createUniqueOptionsFromData(
+  data: Array<Allocator>,
+  key: keyof AllocatorStringValues,
+) {
+  const stringArray = data.map((item) => item[key])
+
+  const counts = countOccurrences(stringArray)
+
+  const sortedUniqueCountries = Object.keys(counts).sort(
+    (a, b) => counts[b] - counts[a],
+  )
+
+  const sortUniqueOptions = sortedUniqueCountries.map((country) => ({
+    id: country,
+    name: country,
+  }))
+
+  return sortUniqueOptions
+}
+
+function countOccurrences(array: Array<string>) {
+  return array.reduce(
+    (object, item) => {
+      const curentCount = object[item] || 0
+      object[item] = curentCount + 1
+      return object
+    },
+    {} as Record<string, number>,
+  )
+}
```

Comment:
```suggestion
  const valuesArray = data.map((item) => item[key])
  const valueCounts = countOccurrences(valuesArray)

  const sortedUniqueOptions = Object.keys(valueCounts)
    .sort((a, b) => valueCounts[b] - valueCounts[a])
    .map((value) => ({
      id: value,
      name: value,
    }))

  return sortedUniqueOptions
}

function countOccurrences(array: Array<string>) {
  return array.reduce((counts, item) => {
    counts[item] = (counts[item] || 0) + 1
    return counts
  }, {} as Record<string, number>)
}
```

Reply:
Sidenote - I'm happy to keep `reduce` but I do think it tends to throw off people. `forEach` can do the same thing.

```
function countOccurrences(array: Array<string>) {
  const counts: Record<string, number> = {}

  array.forEach((item) => {
    if (counts[item]) {
      counts[item] += 1
    } else {
      counts[item] = 1
    }
  })

  return counts
}
```

Reply:
Lastly, what about `generateSortedOptions`?

Reply:
Love it :)